cluster-autoscaler/estimator/pod_priority_processor.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package estimator
+
+import (
+	"sort"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// PodScoreInfo contains Pod and score that corresponds to how important it is to handle the pod first.
+type PodScoreInfo struct {
+	score float64
+	pod   *apiv1.Pod
+}
+
+// PodPriorityProcessor is an interface used to determine the order of the pods
+// used while binpacking during scale up estimation
+type PodPriorityProcessor interface {
```

Comment:
IMO the name is both very generic, and seems misleadingly related to the k8s pod priority concept - which IIUC this interface isn't actually concerned with.

I'd go with one of these instead:
* If it's supposed to be a generic processor for pods at the beginning of binpacking for each node group, I'd omit the "priority" part, and add something about when it's called - e.g. "PreEstimationPodProcessor". But I'm not sure what use cases we'd have for generic processing of these pods at the beginning of binpacking. On the other hand IMO it'd have some conceptual issues - if it's a generic processing hook, the implementation could think it's a good idea to e.g. add pods to the list, which IIUC would not play well with BinpackingEstimator (it assumes all passed pods can be scheduled on the nodes from the node group). Probably not a big issue, more of a nit if you want to go this way.
* If it's supposed to be just for changing the order of processing the pods during binpacking, I'd tie the name to that - e.g. "EstimationPodOrderer". IMO this is a better way to go, as the interface clearly implies that implementations should only be concerned with determining the order of passed pods.

Reply:
Thanks for the detailed comment.
The definite concept behind the process is to just reorder the pods before binpacking instead of being a "generic"  pod processor before binpacking. I also agree having a generic processor before binpacking could be mis-represented and lead to unintended consequences.
I've renamed to `EstimationPodOrderer`.